### ISSUE TITLE
0.9.0 breaks tests with java.lang.NoSuchFieldError 

### ISSUE THREAD

**Issue by aahlenst** [facebook/screenshot-tests-for-android#217]
0.9.0 causes previously working on-device tests to fail:

```
java.lang.NoSuchFieldError: No static field abc_btn_check_material_anim of type I in class Landroidx/appcompat/R$drawable; or its superclasses (declaration of 'androidx.appcompat.R$drawable' appears in /data/app/com.example.myapp.test-c2dXIi1jTYcXHQPnmvkJRQ==/base.apk)
    at androidx.appcompat.widget.AppCompatDrawableManager.<clinit>(AppCompatDrawableManager.java:174)
```

**Comment by lwasyl** [comment]
Similar issue happened to me, but it couldn't find `abc_vector_test` resource (`android.content.res.Resources$NotFoundException: Resource ID #0x7f090053`). The resource is in the apk, but the ID didn't match one that was expected. 

I believe it's some version mismatch, but I couldn't figure out where, I tried pinning all dependencies to one version for both `implementation` and `androidTestImplementation` with no luck. But also I remember I had an impression that `0.8.0` didn't pull any dependencies at all, while `0.9.0` pulls a bunch of app compat artifacts. I tried updating appcompat to 28.0.0 which should be more compatible with androidx, but it didn't work either.

Sidenote, that's why I submitted #216, to make it easier to exclude core library transitive dependencies

**Comment by ewoks** [comment]
any progress?

**Comment by rcd27** [comment]
The same here.

**Comment by gajayraghav** [comment]
I have the same issue. 

android.content.res.Resources$NotFoundException: Drawable com........sampleproject:id/end with resource ID #0x7f090053

i'm using the latest stable releases of androidX with 0.9.0.

**Comment by xiphirx** [comment]
Hey, sorry for the delay. I'm acknowledging the issue, and I think @lwasyl is correct here in that we're likely shipping appcompat artifacts unintentionally. I cannot provide a timeline on when I expect to get to this though, unfortunately :(

**Comment by lwasyl** [comment]
@xiphirx Would it be possible to at least release 0.9.1 with https://github.com/facebook/screenshot-tests-for-android/pull/216 in, so that we can try and workaround the issue locally by excluding transitive dependencies? I'm not entirely sure how big acessiility API changes are across AppCompat versions, but it would allow us to at least try. I can look into this issue again but I'm also not sure when

**Comment by xiphirx** [comment]
I have commits coming in that fix this and the broken tests and the broken sample app, stay tuned. I'll see if we're good for a 0.10.0 release. There were some pretty big underlying changes to how we save images on the device / pull them.

**Comment by xiphirx** [comment]
Before I start to evaluating whether I should package up a new release, do any of you mind pulling the latest snapshot `0.10.0-SNAPSHOT` from https://oss.sonatype.org/content/repositories/snapshots/com/facebook/testing/screenshot/ and seeing if your issue has been resolved?

**Comment by aahlenst** [comment]
Thanks for the work. `0.10.0-SNAPSHOT` looks good, at least locally. Initially, I got:

```
java.lang.NoSuchMethodError: No virtual method getActionList()Ljava/util/List; in class Landroidx/core/view/accessibility/AccessibilityNodeInfoCompat; or its super classes (declaration of 'androidx.core.view.accessibility.AccessibilityNodeInfoCompat' appears in /data/app/com.example.app-F3NxPQTk-GoiEE5zoQMuzw==/base.apk)
at com.facebook.testing.screenshot.layouthierarchy.AccessibilityHierarchyDumper.dumpHierarchy(AccessibilityHierarchyDumper.java:43)
at com.facebook.testing.screenshot.layouthierarchy.AccessibilityHierarchyDumper.dumpHierarchy(AccessibilityHierarchyDumper.java:162)
at com.facebook.testing.screenshot.internal.ScreenshotImpl.record(ScreenshotImpl.java:271)
at com.facebook.testing.screenshot.internal.RecordBuilderImpl.record(RecordBuilderImpl.java:152)
```

A very broad `-keep class androidx.core.view.accessibility.** { *; }` (I'm not in the mood for ProGuard today) in the ProGuard config fixed that and now everything's green again.

**Comment by lwasyl** [comment]
The initial dependency issue seems to be resolved, _however_ for me a test on API 24 fails with
```java
java.lang.IndexOutOfBoundsException: Index: 1, Size: 0
at java.util.ArrayList.get(ArrayList.java:411)
at android.view.ViewGroup.onInitializeAccessibilityNodeInfoInternal(ViewGroup.java:3116)
at android.view.View.onInitializeAccessibilityNodeInfo(View.java:6541)
at androidx.core.view.ViewCompat.onInitializeAccessibilityNodeInfo(ViewCompat.java:650)
at com.facebook.testing.screenshot.layouthierarchy.AccessibilityUtil.createNodeInfoFromView(AccessibilityUtil.java:169)
at com.facebook.testing.screenshot.layouthierarchy.AccessibilityUtil.access$000(AccessibilityUtil.java:40)
at com.facebook.testing.screenshot.layouthierarchy.AccessibilityUtil$AXTreeNode.<init>(AccessibilityUtil.java:579)
at com.facebook.testing.screenshot.layouthierarchy.AccessibilityUtil.generateAccessibilityTree(AccessibilityUtil.java:558)
at com.facebook.testing.screenshot.layouthierarchy.AccessibilityUtil.generateAccessibilityTree(AccessibilityUtil.java:563)
at com.facebook.testing.screenshot.layouthierarchy.AccessibilityUtil.generateAccessibilityTree(AccessibilityUtil.java:563)
at com.facebook.testing.screenshot.layouthierarchy.AccessibilityHierarchyDumper.dumpHierarchy(AccessibilityHierarchyDumper.java:162)
at com.facebook.testing.screenshot.internal.ScreenshotImpl.record(ScreenshotImpl.java:271)
at com.facebook.testing.screenshot.internal.RecordBuilderImpl.record(RecordBuilderImpl.java:152)
```

That's when calling the following:
```kotlin
fun Activity.recordScreenshot(
    name: String? = null,
    group: String? = null,
    description: String? = null
) {
    // calls .hideTextFieldCursors() on every TextView since blinking cursor caused flaky tests
    Espresso.onIdle { hideCursors() } 

    Screenshot.snapActivity(this).run {
        name?.let { setName("$group-$name") }
        group?.let { setGroup(group) }
        description?.let { setDescription(description) }
        record() 
    }
}
```

Another test failed on API 28 with stacktrace
```java
java.lang.RuntimeException: java.lang.RuntimeException: java.lang.NoSuchMethodException: <init> [class android.content.Context, class android.view.Display]
at com.facebook.testing.screenshot.internal.ScreenshotImpl.runCallableOnUiThread(ScreenshotImpl.java:334)
at com.facebook.testing.screenshot.internal.ScreenshotImpl.storeBitmap(ScreenshotImpl.java:163)
at com.facebook.testing.screenshot.internal.ScreenshotImpl.record(ScreenshotImpl.java:267)
at com.facebook.testing.screenshot.internal.RecordBuilderImpl.record(RecordBuilderImpl.java:152)
```

when calling
```kotlin
private fun setupViewAndRecord(view: View) {
    ViewHelpers.setupView(view)
        .setExactWidthDp(320)
        .layout()
    Screenshot.snap(view)
        .record()
}
```

These may or may not be related, I'll probably not have time this week to investigate more

**Comment by xiphirx** [comment]
Thanks for the report! Yeah we noticed the accessibility index out of bounds exception internally as well, we definitely need to fix that before a proper release.

**Comment by xiphirx** [comment]
https://github.com/facebook/screenshot-tests-for-android/releases/tag/0.10.0 Fix is in 0.10.0 

Please file a new task if you still see issues.

**Comment by KishoreKittu467** [comment]
Error:
java.lang.NoSuchFieldError: No static field abc_btn_check_material_anim of type I in class Landroidx/appcompat/R$drawable; or its superclasses (declaration of 'androidx.appcompat.R$drawable'...

Solution:
- Delete the folder ~/.gradle/caches/modules-2/files-2.1/androidx.appcompat/appcompat/1.1.0
- Invalidate caches and restart
